lib/matplotlib/quiver.py

```diff
@@ -542,7 +548,39 @@ def draw(self, renderer):
         super().draw(renderer)
         self.stale = False
 
+    def set_U(self, U):
+        """Set x direction components of the arrow vectors."""
+        self.set_UVC(U, None, None)
+
+    def set_V(self, V):
+        """Set y direction components of the arrow vectors."""
+        self.set_UVC(None, V, None)
+
+    def set_C(self, C):
+        """Set the arrow colors."""
+        self.set_UVC(None, None, C)
+
     def set_UVC(self, U, V, C=None):
+        """
+        Set the U, V (x and y direction components of the arrow vectors) and
+        C (arrow colors) values of the arrows.
+
+        Parameters
+        ----------
+        U : ArrayLike | None
```

Comment:
```suggestion
        U : ArrayLike | None
```
I think a typing/documentation-expert should comment here, but I think we still do write it a bit more "old school", like
`array-like or None`.

Reply:
Yes, we are pretty consistent in docstrings to use `array-like` (I see one instance, for which I am to blame..., where we have an `ArrayLike` that snuck in, but otherwise all docstrings are `array-like`

Actually, numpy uses `array_like` in docstrings, but I would value self consistency over anything else.